server/tests/tram_stop_reachability/test_tram_stop_reachability.py

```diff
@@ -0,0 +1,194 @@
+            == f"Node with id {node_id} not found in the graph."
+        )
+
+    @pytest.mark.parametrize(("edge", "start_stop", "end_stop"), PATH_TOO_LONG_CASES)
+    def test_check_path_viability_path_too_long(
+        self,
+        edge,
+        start_stop,
+        end_stop,
+        krakow_city_configuration: CityConfiguration,
+        krakow_tram_network_graph: nx.DiGraph,
+        tram_stop_pairs: set[tuple[int, int]],
+    ):
+        # Arrange
+        custom_nodes_dict = {
+            (item.source, item.destination): item.ratio
+            for item in krakow_city_configuration.custom_tram_stop_pair_max_distance_checks
+        }
+        krakow_tram_network_graph.remove_edge(*edge)
+        tram_graph_inspector = TramTrackGraphInspector(krakow_tram_network_graph)
+
+        # Act
+        with pytest.raises(PathTooLongError) as exc_info:
+            for start_id, end_id in tram_stop_pairs:
+                tram_graph_inspector.check_path_viability(
+                    start_id,
+                    end_id,
+                    custom_nodes_dict.get(
+                        (start_id, end_id), krakow_city_configuration.max_distance_ratio
+                    ),
+                )
+
+        # Assert
+        assert (
+            str(exc_info.value)
+            .strip()
+            .startswith(f"Path too long: {start_stop} -> {end_stop}")
+        )
+
+    @pytest.mark.parametrize(("edge", "start_stop", "end_stop"), NO_PATH_FOUND_CASES)
+    def test_check_path_viability_no_path_found(
+        self,
+        edge,
+        start_stop,
+        end_stop,
+        krakow_city_configuration: CityConfiguration,
+        krakow_tram_network_graph: nx.DiGraph,
+        tram_stop_pairs: set[tuple[int, int]],
+    ):
+        # Arrange
+        custom_nodes_dict = {
+            (item.source, item.destination): item.ratio
+            for item in krakow_city_configuration.custom_tram_stop_pair_max_distance_checks
+        }
+        krakow_tram_network_graph.remove_edge(*edge)
+        tram_graph_inspector = TramTrackGraphInspector(krakow_tram_network_graph)
+
+        # Act
+        with pytest.raises(NoPathFoundError) as exc_info:
+            for start_id, end_id in tram_stop_pairs:
+                tram_graph_inspector.check_path_viability(
+                    start_id,
+                    end_id,
+                    custom_nodes_dict.get(
+                        (start_id, end_id), krakow_city_configuration.max_distance_ratio
+                    ),
+                )
+
+        # Assert
+        assert (
+            str(exc_info.value).strip()
+            == f"No path found between stops: {start_stop} -> {end_stop}"
+        )
+
+    def test_shortest_path_between_nodes(
+        self,
+        krakow_tram_network_graph: nx.DiGraph,
+        tram_stop_pairs: set[tuple[int, int]],
+    ):
+        # Arrange
+        tram_graph_insector = TramTrackGraphInspector(krakow_tram_network_graph)
```

Comment:
Please check if this typo isn't present in other places 
```suggestion
        tram_graph_inspector = TramTrackGraphInspector(krakow_tram_network_graph)
```

Reply:
Fixed the typo everywhere (I think)